CHANGELOG.adoc

```diff
@@ -2,6 +2,16 @@
 
 == DFX
 
+=== feat!: Remove the wallet proxy and the --no-wallet flag
+
+Breaking change: Canister commands, except for `dfx canister create`, will make the call directly, rather than via the user's wallet. The `--no-wallet` flag is thus removed from `dfx canister` as its behavior is the default.
+
+When working with existing canisters, use the `--wallet` flag in conjunction with `dfx identity get-wallet` in order to restore the old behavior. To upgrade an existing canister to work seamlessly with the new system, the following command should be used (substituting `canister_name` for your own):
+[source, bash]
+----
+dfx canister --wallet "$(dfx identity get-wallet)" update-settings <canister_name> --add-controller "$(dfx identity get-principal)"
```

Comment:
It seems like a lot to ask every developer to do this individually for each of their canisters.

Also, there's a little more to it for canisters that are on the ic network (since the local network is the default):
```
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)"
 update-settings <canister_name> --add-controller "$(dfx identity get-principal)"
```
And even more if they use a different identity on the ic that they do for local development:
```
dfx --identity icident canister --network ic
 --wallet "$(dfx --identity icident identity --network ic get-wallet)"
 update-settings <canister_name>
 --add-controller "$(dfx --identity icident identity get-principal)"
```

Can you think of a way to simplify the upgrade process?

Reply:
Should I include both of those commands? That's a lot to edit in a paste, and they presumably know how the flags work already. Re: simplification, I'm not sure if you're asking for a shell solution or a DFX solution.

Reply:
Just to follow up after our team meeting: let's make sure to include the specific command to run for mainnet canisters:
```
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)"
 update-settings <canister_name> --add-controller "$(dfx identity get-principal)"
```
Also, I think we should call out that `dfx --network ic wallet upgrade` is necessary.

Reply:
Actually we can streamline it a little bit already, using the `--all` parameter rather than calling once with each canister name:
```
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)"
 update-settings --all --add-controller "$(dfx identity get-principal)"
```

Reply:
Should that be for the first command too?

Reply:
Yes I think we should specify `--all` in the instructions for both.